src/components/UserProfile/UserProfileHeaderContent.tsx

```diff
@@ -11,45 +11,44 @@ import {
   Typography,
 } from "@mui/material";
 import { makeStyles, styled } from "@mui/styles";
-import { Theme } from "@mui/system";
 import { SocialInfo } from "../../interfaces/UserInterfaces";
 import { userRoleConvertor } from "../../utils/userRoles";
 
 const CustomTypographyPrimary = styled(Typography)({
-  fontSize: "25px",
-  fontWeight: "500",
-});
-
-const CustomTypographySecondary = styled(Typography)({
-  color: "#818386",
-  fontSize: "25px",
-  fontWeight: "500",
-});
-
-const useStyles = makeStyles((theme: Theme) => ({
-  socialMediaIcon: {
-    "&:hover": {
-      color: "blue",
+    fontSize: "25px",
+    fontWeight: "500",
+  }),
+  CustomTypographySecondary = styled(Typography)({
+    color: "#818386",
+    fontSize: "25px",
+    fontWeight: "500",
+  }),
+  useStyles = makeStyles(() => ({
+    socialMediaIcon: {
+      "&:hover": {
+        color: "blue",
+      },
     },
-  },
-}));
+  }));
 
 interface UserProfileDisplayProps {
   email: string;
   displayName: string;
   profileImageUrl: string;
-  socialInfo: string;
+  socialInfo: SocialInfo | string;
   phone: string;
   roles: number[];
-  switchToEditModeHandler: any;
+  switchToEditModeHandler?: any;
+  children?: JSX.Element | JSX.Element[];
 }
 
-const CurrentUserProfileHeaderContent = (props: UserProfileDisplayProps) => {
-  const classes = useStyles();
-
-  const socialInfo: SocialInfo = props.socialInfo
-    ? JSON.parse(props.socialInfo)
-    : { facebook: "", linkedIn: "" };
+const UserProfileHeaderContent = (props: UserProfileDisplayProps) => {
+  const socialInfo: SocialInfo = !props.socialInfo
+      ? { facebook: "", linkedIn: "" }
+      : typeof props.socialInfo === "string"
+      ? JSON.parse(props.socialInfo as string)
+      : props.socialInfo,
+    classes = useStyles();
```

Comment:
What if we destructure the props and give it the default value `{ facebook: "", linkedIn: ""}`, like this:

```js
const UserProfileHeaderContent: React.FC<UserProfileDisplayProps> = ({
  socialInfo = { facebook: "", linkedIn: "" },
  ...props
})
```

I don't get where should that string come from, I may be wrong, but all this `? : ? :` is really hard to read

Reply:
I think that your proposal would make the code more readable, however it doesn't properly handle the case when the socialInfo field comes as null from the back-end :(

Reply:
Btw, the ES linter doesn't do a great job at formatting this code block.

Reply:
```suggestion
  const parseSocialInfo =
      typeof props.socialInfo === "string"
        ? JSON.parse(props.socialInfo as string)
        : props.socialInfo,
    socialInfo: SocialInfo = !props.socialInfo
      ? { facebook: "", linkedIn: "" }
      : parseSocialInfo,
    classes = useStyles();
```

What if we do it like this? Other than this it looks good to me

Reply:
Looks better indeed, I will apply your suggestion, thanks!